src/ooxml/testcases/org/apache/poi/xssf/usermodel/TestXSSFSheetShiftColumns.java

```diff
@@ -0,0 +1,433 @@
+        // comment in column 0 is gone
+        comment = sheet.getCellComment(new CellAddress(0, 0));
+        assertNull(comment);
+
+        // comment is column in column 1
+        comment = sheet.getCellComment(new CellAddress(0, 1));
+        assertNotNull(comment);
+        assertEquals("Amdocs", comment.getAuthor());
+        assertEquals("Amdocs:\ntest\n", comment.getString().getString());
+
+        Workbook wbBack = XSSFTestDataSamples.writeOutAndReadBack(wb);
+        wb.close();
+        assertNotNull(wbBack);
+
+        Sheet sheetBack = wbBack.getSheetAt(0);
+
+        // comment in column 0 is gone
+        comment = sheetBack.getCellComment(new CellAddress(0, 0));
+        assertNull(comment);
+
+        // comment is now in column 1
+        comment = sheetBack.getCellComment(new CellAddress(0, 1));
+        assertNotNull(comment);
+        assertEquals("Amdocs", comment.getAuthor());
+        assertEquals("Amdocs:\ntest\n", comment.getString().getString());
+        wbBack.close();
+    }
+    
+    // transposed version of TestXSSFSheetShiftRows.testBug54524()
+    @Test
+    public void testBug54524() throws IOException {
+        Workbook wb = _testDataProvider.createWorkbook();
+        Sheet sheet = wb.createSheet();
+        Row firstRow = sheet.createRow(0); 
+        firstRow.createCell(0).setCellValue("");
+        firstRow.createCell(1).setCellValue(1);
+        firstRow.createCell(2).setCellValue(2);
+        firstRow.createCell(3).setCellFormula("SUM(B1:C1)");
+        firstRow.createCell(4).setCellValue("X");
+        
+        sheet.shiftColumns(3, 5, -1);
+
+        Cell cell = CellUtil.getCell(sheet.getRow(0), 1);
+        assertEquals(1.0, cell.getNumericCellValue(), 0);
+        cell = CellUtil.getCell(sheet.getRow(0), 2);
+        assertEquals("SUM(B1:B1)", cell.getCellFormula());
+        cell = CellUtil.getCell(sheet.getRow(0), 3);
+        assertEquals("X", cell.getStringCellValue());
+        wb.close();
+    }
+
+    // I need these methods for testing. When we finish with project, we can easily remove them. 
+    // Dragan Jovanović
+    public static void writeSheetToLog(Sheet sheet) {
+        int rowIndex = sheet.getFirstRowNum();
+        while (rowIndex <= sheet.getLastRowNum()) {
+            Row row = sheet.getRow(rowIndex);
+            if (row == null)
+                //log.trace("null row");
+                System.out.println("null row");
+            else {
+                String line = "";
+                for(int columnIndex = 0; columnIndex < 5; columnIndex++){
+                    //String comment = sheet.getCellComment(new CellAddress(rowIndex, columnIndex)) == null ? "no comment" : sheet.getCellComment(new CellAddress(rowIndex, columnIndex)).getString().getString();
+                    //line +=  String.format("; %1$12s    %2$20s",  ""/*getValue(row.getCell(columnIndex))*/, comment);
+                    line +=  String.format("; %1$12s",  getValue(row.getCell(columnIndex)));
```

Comment:
Use `DataFormatter#formatCellValue(Cell)` instead of writing your own switch statement.
https://poi.apache.org/apidocs/org/apache/poi/ss/usermodel/DataFormatter.html#formatCellValue-org.apache.poi.ss.usermodel.Cell-

There's a working example here: https://poi.apache.org/spreadsheet/quick-guide.html#Getting+the+cell+contents